File: Cpp/src/array/sum_of_1d_array.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#include "sum_of_1d_array.h"


using namespace std;

vector<int> SumOf1DArray::RunningSum(vector<int> &nums) {
    for (int i = 1; i < nums.size(); i++) {
        nums[i] += nums[i - 1];
    }
    return nums;
}

File: Kotlin/src/main/kotlin/com/github/msorkhpar/leetcodelearningcards/array/SumOf1DArray.kt
package com.github.msorkhpar.leetcodelearningcards.array

class SumOf1DArray {
    fun runningSum(nums: IntArray): IntArray {
        for (index in 1 until nums.size) {
            nums[index] += nums[index - 1]
        }
        return nums
    }
}